src/routes/manage_job/post.rs

```diff
@@ -0,0 +1,264 @@
+				port: proxy.get("port").unwrap().parse().unwrap(),
+			});
+		}
+
+		input
+	}
+}
+
+/// Endpoint response body.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct CreateBulkResponseBody {
+	job_id: i32,
+}
+
+// Arguments to the `#[job]` attribute allow setting default job options.
+/// This task tries to verify the given email and inserts the results
+/// into the email verification db table
+#[job]
+pub async fn email_verification_task(
+	mut current_job: CurrentJob,
+	// Additional arguments are optional, but can be used to access context
+	// provided via [`JobRegistry::set_context`].
+) -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
+	let task_input: TaskInput = current_job.json()?.unwrap();
+
+	// Retry each future twice, to avoid grey-listing.
+	if let Ok((response, _)) = retry(&task_input.input, RetryOption::Direct, 2).await {
+		let rec = sqlx::query!(
+			r#"
+			INSERT INTO email_results (job_id, result)
+			VALUES ($1, $2)
+			"#,
+			task_input.job_id,
+			serde_json::json!(response)
+		);
+	} else {
+		log::debug!(
+			target:"reacher",
+			"Failed [email={}] for [job={}]",
+			task_input.job_id,
+			task_input.input.to_emails[0]
+		);
+	}
+
+	current_job.complete().await?;
+
+	Ok(())
+}
+
+/// Retry the check ciee_check_email function, in particular to avoid
+/// greylisting.
+#[async_recursion]
+async fn retry(
+	input: &CheckEmailInput,
+	retry_option: RetryOption,
+	count: usize,
+) -> Result<(CheckEmailOutput, RetryOption), CheckEmailError> {
+	log::debug!(
+		target:"reacher",
+		"[email={}] Checking with retry option {}, attempt #{}",
+		input.to_emails[0],
+		retry_option,
+		count,
+	);
+
+	let result = ciee_check_email(input)
+		.await
+		.pop()
+		.expect("Input contains one email, so does output. qed.");
+
+	log::debug!(
+		target:"reacher",
+		"[email={}] Got result with retry option {}, attempt #{}, is_reachable={:?}",
+		input.to_emails[0],
+		retry_option,
+		count,
+		result.is_reachable
+	);
+
+	// If we get an unknown error, log it.
+	known_errors::log_unknown_errors(&result, retry_option);
+
+	if result.is_reachable == Reachable::Unknown {
+		if count <= 1 {
+			Err(CheckEmailError::Unknown((result, retry_option)))
+		} else {
+			retry(input, retry_option, count - 1).await
+		}
+	} else {
+		Ok((result, retry_option))
+	}
+}
+
+/// handles input, creates db entry for job and tasks for verification
+async fn create_bulk_request(
+	body: CreateBulkRequestBody,
+	conn_pool: Pool<Postgres>,
+) -> Result<impl warp::Reply, warp::Rejection> {
+	// create job entry
+	let rec = sqlx::query!(
+		r#"
+		INSERT INTO bulk_jobs (total_records)
+		VALUES ($1)
+		RETURNING id
+		"#,
+		body.input.len() as i32
+	)
+	.fetch_one(&conn_pool)
+	.await
+	.map_err(|e| {
+		log::error!(
+			target:"reacher/v0/bulk/",
+			"Failed to create job record for [body={:?}] with [error={}]",
+			&body,
+			e
+		);
+		ReacherError::from(e)
+	})?;
+
+	for job in body.into_iter() {
```

Comment:
I'm not sure I see the value of `CreateBulkRequestBodyIterator`

Can we do:
```rust 
	let inputs: Vec<CheckEmailInput> = body.into();
	for job in inputs.into_iter() {/*same*/}
```

We just need to add a:

```
impl Into<Vec<CheckEmailInput>> for CreateBulkRequestBody {}
```

which should replace the `impl Into<CheckEmailInput> for CreateBulkRequestBody` on L89, which seems incorrect to me: we should consider one CheckEmailInput == one email, whereas the CreateBulkRequestBody contains multiple emails.

In this case, we can remove `CreateBulkRequestBodyIterator`

Reply:
I think we do need an iterator. Suppose there's a bulk job with 100,000 individual tasks, the CreateRequest will only be one object but it'll get converted to 100,000 `CheckEmailInput` objects. Rather than doing this whole thing conversion in-memory at once, It's almost always better to iteratively keep converting them and passing them on to the task queue.

Reply:
As discussed we're going ahead with the iterator style, just changing the item of the iterator to be a `CheckEmailInput`, do the iterating over the request directly gives task inputs.